debezium-core/src/main/java/io/debezium/jdbc/JdbcValueConverters.java

```diff
@@ -976,6 +985,8 @@ else if (data instanceof Float)
             decimal = BigDecimal.valueOf(((Float) data).doubleValue());
         else if (data instanceof Double)
             decimal = BigDecimal.valueOf(((Double) data).doubleValue());
+        else if (data instanceof SpecialValue)
+            return null;
```

Comment:
That part is concerning me a bit:

* I don't think it will work if the column isn't nullable
* Representing `Infinity` as null seems weird

How about this:

* We add a new schema type, `SpecialValueDecimal`; this resembles Kafka's `Decimal` type but has another, optional `String` field whose value can be one of "NaN", "Infinity" or "-Infinity". If it is set, the value will be 0 (or even null).
* We also add such optional field to `VariableScaleDecimal`, allowing to represent those special values for variable precision decimals.
* We add another value to `PRECISE_WITH_SPECIAL_VALUES` to the `DecimalMode` option; if this value is chosen, we'd encode decimals using `SpecialValueDecimal`.

That way people will have to opt into getting special value support with precise decimal mapping and the value structure is a bit more complex, but it seems like a safer option and more versatile to me. WDYT?

Reply:
@gunnarmorling Hmm strange, I've a feeling I've already seen it somewhere ;-)
https://issues.jboss.org/browse/DBZ-606?focusedCommentId=13535252&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13535252